confignode/src/main/java/org/apache/iotdb/confignode/client/SyncDataNodeClientPool.java

```diff
@@ -88,19 +88,21 @@ public TSStatus invalidateSchemaCache(
 
   public void deleteRegions(Set<TRegionReplicaSet> deletedRegionSet) {
     Map<TDataNodeLocation, List<TConsensusGroupId>> regionLocationMap = new HashMap<>();
-    deletedRegionSet.forEach(
-        (tRegionReplicaSet) -> {
-          final List<TDataNodeLocation> dataNodeLocations =
-              tRegionReplicaSet.getDataNodeLocations();
-          regionLocationMap
-              .computeIfAbsent(dataNodeLocations.get(0), k -> new ArrayList<>())
-              .add(tRegionReplicaSet.getRegionId());
-        });
-    LOGGER.info("Current regionLocationMap {} ", regionLocationMap);
-    regionLocationMap.forEach(
-        (dataNodeLocation, regionIds) -> {
-          deleteRegions(dataNodeLocation.getInternalEndPoint(), regionIds, deletedRegionSet);
-        });
+    synchronized (deletedRegionSet) {
```

Comment:
Not sure the `synchronized` would take effect or not because the `deletedRegionSet` is the parameter of the method

Reply:
Now I synchronized in PartitionInfo.getDeletedRegionSet()